packages/engine/src/ecs/classes/EngineService.ts

```diff
@@ -55,11 +55,25 @@ export function EngineEventReceptor(action: EngineActionType) {
       case EngineEvents.EVENTS.SCENE_UNLOADED:
         return s.merge({ sceneLoaded: false, sceneLoading: false })
       case EngineEvents.EVENTS.SCENE_LOADING:
-        return s.merge({ sceneLoaded: false, sceneLoading: true })
-      case EngineEvents.EVENTS.SCENE_LOADED:
-        return s.merge({ sceneLoaded: true, sceneLoading: false })
-      case EngineEvents.EVENTS.JOINED_WORLD:
-        return s.merge({ joinedWorld: true })
+        return s.merge({
+          sceneLoaded: false,
+          sceneLoading: true,
+          loadingProgress: 0,
+          loadingDetails: 'common:loader.loadingObjects'
+        })
+      case EngineEvents.EVENTS.SCENE_LOADED: {
+        const message = s.joinedWorld.value ? 'common:loader.loadingComplete' : 'common:loader.joiningWorld'
+        return s.merge({ sceneLoaded: true, sceneLoading: false, loadingProgress: 100, loadingDetails: message })
```

Comment:
i would advise against pushing i18n into the engine package, this is a property of the client rather than the engine itself, we should have a LoadingStage enum or something that the client can then interpret

Reply:
Good idea